lib/ex_unit/lib/ex_unit/formatter.ex

```diff
@@ -186,10 +186,18 @@ defmodule ExUnit.Formatter do
 
   defp inspect_multiline(expr, width) do
     expr
-    |> inspect(pretty: true, width: width)
+    |> safe_inspect(pretty: true, width: width)
     |> String.replace("\n", "\n" <> @inspect_padding)
   end
 
+  defp safe_inspect(expr, opts \\ []) do
+    try do
+      inspect(expr, opts)
+    rescue
+      e -> "inspect failed: #{inspect e}"
```

Comment:
If it fails, you can call it again but passing `structs: false` (it is the only reason it can fail).